src/macro-context.js

```diff
@@ -184,9 +196,49 @@ export default class MacroContext {
       default:
         throw new Error('Unknown term type: ' + type);
     }
+
+    Object.assign(this[symShadow], advance(enf, this[symShadow]));
     return {
       done: false,
       value: new SyntaxOrTermWrapper(value, this.context),
     };
   }
+
+  prev() {
+    let shadow = this[symShadow];
+    let resetRestCount = this[symResetVals].rest.size;
+    if(resetRestCount > shadow.rest.size) {
+      let enf = this[symEnf];
+      Object.assign(shadow, recede(shadow));
+      Object.assign(this[symEnf], { term: null, rest: shadow.rest });
```

Comment:
@disnet: Maybe the enforester could ask the macro context for the most up-to-date `rest` list after the transform returns?